helix-term/src/commands/lsp.rs

```diff
@@ -452,43 +452,83 @@ pub fn code_action(cx: &mut Context) {
     cx.callback(
         future,
         move |editor, compositor, response: Option<lsp::CodeActionResponse>| {
-            let actions = match response {
+            let mut actions = match response {
                 Some(a) => a,
                 None => return,
             };
+
             if actions.is_empty() {
                 editor.set_status("No code actions available");
                 return;
             }
 
-            let mut picker = ui::Menu::new(actions, (), move |editor, code_action, event| {
-                if event != PromptEvent::Validate {
-                    return;
-                }
+            // sort by CodeActionKind
+            // this ensures that the most relevant codeactions (quickfix) show up first
+            // while more situational commands (like refactors) show up later
+            // this behaviour is modeled after the behaviour of vscode (editor/contrib/codeAction/browser/codeActionWidget.ts)
+
+            let mut categories = vec![Vec::new(); 8];
+            for action in actions.drain(..) {
```

Comment:
Couldn't this just use `sort_by`/`sort_by_key`?

Reply:
Oh yeah good catch, I modeled this after the original implementation in vscode but a stable sort is a better fit here. I have updated the PR to use `sort_by_key`